lib/browser/api/menu-item.js

```diff
@@ -7,18 +7,13 @@ let nextCommandId = 0
 const MenuItem = function (options) {
   const {Menu} = require('electron')
 
-  this.selector = options.selector
-  this.type = options.type
-  this.role = options.role
-  this.label = options.label
-  this.sublabel = options.sublabel
-  this.accelerator = options.accelerator
-  this.icon = options.icon
-  this.enabled = options.enabled
-  this.visible = options.visible
-  this.checked = options.checked
-
-  this.submenu = options.submenu
+  // Clone and sanitize the provided options
+  options = Object.assign({}, options)
+  delete options.overrideProperty
+  delete options.overrideReadOnlyProperty
```

Comment:
I don't think we want the options object to override any already defined instance properties or methods here, so maybe the `hasOwnProperty` guard that was previously used in `Menu.buildFromTemplate` is a better approach.